File: go/create-target-array-in-the-given-order.go
func createTargetArray(nums []int, index []int) []int {
    targetArray := make([]int, 0)
    for i := 0; i < len(nums); i++ {
        newTargetArray := make([]int, len(targetArray) + 1)
        copy(newTargetArray[:index[i]], targetArray[:index[i]])
        newTargetArray[index[i]] = nums[i]
        copy(newTargetArray[index[i] + 1:], targetArray[index[i]:])
        targetArray = newTargetArray
    }
    return targetArray
}

File: python/create-target-array-in-the-given-order.py
class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        target_array = []
        n = len(nums)
        for i in range(n):
            #target_array = target_array[:index[i]] + [nums[i]] + target_array[index[i]:]
            target_array.insert(index[i], nums[i])
        return target_array
